### PR TITLE
Switch from Webdriverio to Puppeteer

### PR DESCRIPTION
**Pull request by domoritz**
Advantages

* wdio v5 does not support the static server that we need
* We could not use jest, now we can
* fewer dependencies

### CODE REVIEW

**Comment by domoritz** [comment]
Tests are failing because we use different ids to clip: https://www.diffchecker.com/1lSThiFF.

**Comment by domoritz** [comment]
@arvind @jheer @kanitw How is it possible that tests pass in node 11 but not before with the failures being due to different SVGs? Shouldn't the SVG generation be predictable/repeatable in different node versions? 

screen shot 2019-01-23 at 22 18 49

**Comment by kanitw** [comment]
That’s why I disabled it lol

**Comment by domoritz** [comment]
Not a fan :-P I changed it so we run runtime tests only on node 11.

**Comment by kanitw** [comment]
> Not a fan :-P I changed it so we run runtime tests only on node 11.

I believe we didn't run Node 11 back then so it wasn't an option.

**Comment by domoritz** [comment]
It must have been correct on some version of node, no?

**Comment by kanitw** [comment]
> It must have been correct on some version of node, no?

Right now just hope for node 11.   :) 


LGTM, but I'll let @arvind merge this.

File: test-runtime/resolve.test.ts [hunk 1]
```diff
@@ -15,37 +14,45 @@ import {
...
+        it('should have one global selection instance', async () => {
           const selection = {
             type,
             resolve: 'global',
             ...(specType === 'facet' ? {encodings: ['y']} : {})
           };
 
           for (let i = 0; i < hits[specType].length; i++) {
-            embed(spec(specType, i, selection));
+            await await embed(spec(specType, i, selection));
```
Comment (arvind) [inline test-runtime/resolve.test.ts]: `await await`?

Comment (domoritz) [reply test-runtime/resolve.test.ts]: Ups

Comment (domoritz) [reply test-runtime/resolve.test.ts]: Fixed in https://github.com/vega/vega-lite/pull/4506

**Pull request merged**